### PR TITLE
@craigspaeth - Add guests and submit

### PR DESCRIPTION
**Pull request by broskoski**
rsvp-initial

### CODE REVIEW

File: apps/event/controllers/index.js [hunk 1]
```diff
@@ -7,14 +8,50 @@ const api = new Lokka({
...
+    name: '',
+    email: '',
+    event_id: ctx.params.id,
+    guests: []
+  })
   ctx.render({ body: Index })
 }
+
+export const setNumOfGuests = async (num) => {
+  state.set('num_of_guests', num)
```
Comment (craigspaeth) [inline apps/event/controllers/index.js]: Snake case!?!?!?!?!??!?!?!?!??!!? CamelCase all the things!!!!!!!!! 💢  🔥  😠  🔥  💢 

jkjk, as long as you're consistent. Being an all JS stack I've found kebab-case for filenames, ClassCase for classes/prototypes, and camelCase for everything else to be the nicest frame-of-mind. Totally a personal style thing though, but thought I'd call it out for the sake of debate.

File: apps/event/controllers/index.js [hunk 2]
```diff
@@ -7,14 +8,50 @@ const api = new Lokka({
...
+}
+
+export const createReservation = async (e) => {
+  e.preventDefault()
+  let guests = []
+  times(state.get('num_of_guests'), (index) => {
```
Comment (craigspaeth) [inline apps/event/controllers/index.js]: `times` can be used as a `map` of sorts too so you could use `const guests = times(...)` to be all non-mutative and such.

File: apps/event/models/reservation/index.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { model, string, array, objectid, date, number } from 'joiql-mongo'
+
+export default model('reservation', {
+  name: string(),
+  email: string(),
+  guests: array().items(string()),
+  event_id: string(),
+  created_at: date()
+    .meta((is) => ({
```
Comment (craigspaeth) [inline apps/event/models/reservation/index.js]: I've been thinking of a [better dsl for this validation](https://github.com/muraljs/joiql-mongo/issues/2) if you get a chance lmk your thoughts.

File: apps/event/views/index.js [hunk 4]
```diff
@@ -32,6 +39,17 @@ view.styles({
       padding: 5,
       outline: 'none'
     }
+  ),
+  button: assign(
+    type('avantgarde', 'mediumHeadline'),
```
Comment (craigspaeth) [inline apps/event/views/index.js]: Now that we have at-least two apps using this way inline-styling we should think about extracting a JS-only Elan of sorts (or moving to Elan 💭 )!

**Comment by craigspaeth** [comment]
👏  very cool! I know this is time constrained, so I won't hold the PR up for it—but I challenge you to write tests. You can see [how we've done it in Team Nav](https://github.com/artsy/team-navigator/tree/v2/test).

File: apps/event/views/index.js [hunk 5]
```diff
@@ -41,12 +59,39 @@ view.render(() =>
...
+      (() => {
+        let els = []
+        times(state.get('num_of_guests'), (index) => {
+          els.push([
+            label('.label', `Your Guests Name`),
+            input('.input', { name: `guests[${index}]` })
+          ])
+        })
+        return els
+      })(),
```
Comment (craigspaeth) [inline apps/event/views/index.js]: Nice use of IIFE! I'll re-point out `times` being an expression already though.

**Comment by joeyAghion** [comment]
Woah! Just wondering if we evaluated any of the many white-label event/RSVP systems out there before starting this?



Also, this repo name actually communicates the project purpose... what were you thinking?

**Comment by broskoski** [comment]
Yep, we definitely evaluated other options :)

We ultimately decided that its best to build our own because (besides all of the specific needs that we not always completely covered by the white label services) it makes us easier to integrate RSVP-ing into user accounts if we control the service completely.

File: apps/event/controllers/index.js (hunk 1 above)
Comment (broskoski) [inline apps/event/controllers/index.js]: Yeah, I don't know what it is, I always feel like db fields or attribute names should be snake case! CamelCase is for classes, functions, etc.

**Pull request merged**